packages/components/src/form-token-field/test/index.tsx

```diff
@@ -0,0 +1,2089 @@
+
+	expect( allVisibleOptions ).toHaveLength( suggestionsText.length );
+
+	allVisibleOptions.forEach( ( matchedOption, index ) => {
+		expect( matchedOption ).toHaveAccessibleName(
+			suggestionsText[ index ]
+		);
+	} );
+};
+
+function unescapeAndFormatSpaces( str: string ) {
+	const nbsp = String.fromCharCode( 160 );
+	const escaped = new DOMParser().parseFromString( str, 'text/html' );
+	return escaped.documentElement.textContent?.replace( / /g, nbsp ) ?? '';
+}
+
+describe( 'FormTokenField', () => {
+	describe( 'basic usage', () => {
+		it( "should tokens with the input's value when pressing the enter key", async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
+			} );
+
+			const onChangeSpy = jest.fn();
+
+			render( <FormTokenFieldWithState onChange={ onChangeSpy } /> );
+
+			const input = screen.getByRole( 'combobox' );
+
+			// Add 'apple' token by typing it and pressing enter to tokenize it.
+			await user.type( input, 'apple[Enter]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 1 );
+			expect( onChangeSpy ).toHaveBeenCalledWith( [ 'apple' ] );
+			expectTokensToBeInTheDocument( [ 'apple' ] );
+
+			// Add 'pear' token by typing it and pressing enter to tokenize it.
+			await user.type( input, 'pear[Enter]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 2 );
+			expect( onChangeSpy ).toHaveBeenLastCalledWith( [
+				'apple',
+				'pear',
+			] );
+			expectTokensToBeInTheDocument( [ 'apple', 'pear' ] );
+		} );
+
+		it( "should add a token with the input's value when pressing the comma key", async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
+			} );
+
+			const onChangeSpy = jest.fn();
+
+			render( <FormTokenFieldWithState onChange={ onChangeSpy } /> );
+
+			const input = screen.getByRole( 'combobox' );
+
+			// Add 'orange' token by typing it and pressing enter to tokenize it.
+			await user.type( input, 'orange,' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 1 );
+			expect( onChangeSpy ).toHaveBeenCalledWith( [ 'orange' ] );
+			expectTokensToBeInTheDocument( [ 'orange' ] );
+		} );
+
+		it( 'should add a token with the input value when pressing the space key and the `tokenizeOnSpace` prop is `true`', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
+			} );
+
+			const onChangeSpy = jest.fn();
+
+			const { rerender } = render(
+				<FormTokenFieldWithState onChange={ onChangeSpy } />
+			);
+
+			const input = screen.getByRole( 'combobox' );
+
+			// Add 'dragon fruit' token by typing it and pressing enter to tokenize it.
+			await user.type( input, 'dragon fruit[Enter]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 1 );
+			expect( onChangeSpy ).toHaveBeenCalledWith( [ 'dragon fruit' ] );
+			expectTokensToBeInTheDocument( [ 'dragon fruit' ] );
+
+			rerender(
+				<FormTokenFieldWithState
+					onChange={ onChangeSpy }
+					tokenizeOnSpace={ true }
```

Comment:
We could omit `={ true }`

```suggestion
					tokenizeOnSpace
```